proxy.go

```diff
@@ -142,7 +142,9 @@ func (p *Proxy) Close() {
 	close(p.closing)
 
 	log.Infof("martian: waiting for connections to close")
```

Comment:
Ideally, wouldn't `Close` wait until all of the outstanding calls to `Serve` have returned? IIUC, since `Listener.Accept` is blocking, the best way to do that would be to keep a slice of `net.Listener`s being used by `Serve`, and close them all during `Close`. And maybe another `sync.WaitGroup` to keep track of the `Serve` calls. Let me know if you think this is a good change, it's independent from this PR.